src/main/java/com/iota/iri/cache/CacheConfiguration.java

```diff
@@ -0,0 +1,128 @@
+package com.iota.iri.cache;
+
+/**
+ * Cache configuration. This class represents the config values used to instantiate a {@link Cache}
+ */
+public class CacheConfiguration {
+
+    public static final long DEFAULT_MAX_SIZE = 256;
+    public static final int DEFAULT_EVICTION_COUNT = 10;
+    public static final boolean DEFAULT_WEAK_REFERENCE = false;
+    public static final CacheEvictionPolicy DEFAULT_EVICTION_POLICY = CacheEvictionPolicy.FIFO;
+    public static final boolean DEFAULT_IS_NULL_ALLOWED = false;
+    public static final int DEFAULT_TIME_TO_IDLE_SECONDS = 0;
+
+    /**
+     * Max size of the cache
+     */
+    private long maxSize = DEFAULT_MAX_SIZE;
+
+    /**
+     * The number of items to evict in a single batch of eviction.
+     */
+    private int evictionCount = DEFAULT_EVICTION_COUNT;
+
+    /**
+     * A flag to determine if the cache should use weak/soft reference to values or not
+     */
+    private boolean weakReference = DEFAULT_WEAK_REFERENCE;
+
+    /**
+     * The maximum number of seconds an element can exist in cache without it being accessed. Idle elements will not be
+     * returned. The default value is 0, which means there's no TTL on this cache
+     */
+    private int timeToIdleSeconds = DEFAULT_TIME_TO_IDLE_SECONDS;
+
+    /**
+     * The {@link CacheEvictionPolicy} policy used.
+     */
+    private CacheEvictionPolicy evictionPolicy = DEFAULT_EVICTION_POLICY;
+
+    /**
+     * Determines if null values are allowed in the cache
+     */
+    private boolean isNullAllowed = DEFAULT_IS_NULL_ALLOWED;
```

Comment:
When you create the interface have the java docs on the public methods (like the bot required).

You may keep java docs on private fields, but I don't require it. In fact I think it is extraneous because you have to fix the docs twice